### PR TITLE
Feature/ktlint

### PR DESCRIPTION
**Pull request by avidraghavMM**

### CODE REVIEW

File: app/src/main/java/com/raghav/spacedawn/ui/AppViewModelFactory.kt [hunk 1]
```diff
@@ -9,7 +9,7 @@ class AppViewModelFactory(
     private val app : Application,
     private val repository: AppRepository
 ): ViewModelProvider.Factory {
-    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
+    override fun <T : ViewModel> create(modelClass: Class<T>): T {
```
Comment (aditya-bhawsar-mm) [inline app/src/main/java/com/raghav/spacedawn/ui/AppViewModelFactory.kt:12]: i would rather prefer to have this done with help of dagger

File: .idea/gradle.xml [hunk 2]
```diff
@@ -4,7 +4,7 @@
   <component name="GradleSettings">
```
Comment (aditya-bhawsar-mm) [inline .idea/gradle.xml:4]: lets add all the `.idea` files to `.gitignore` as we dont need them

**Review by aditya-bhawsar-mm — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/raghav/spacedawn/ui/AppViewModelFactory.kt (hunk 1 above)
Comment (avidraghavMM) [reply app/src/main/java/com/raghav/spacedawn/ui/AppViewModelFactory.kt:12]: Agree, currently the project doesn't has any DI framework implemented. Should I make another PR for adding Dagger or should I do that in this PR itself? 
Also can you attach appropriate resources for setting-up/adding Dagger to an android project?

File: .idea/gradle.xml (hunk 2 above)
Comment (avidraghavMM) [reply .idea/gradle.xml:4]: sure.

File: app/src/main/java/com/raghav/spacedawn/ui/AppViewModelFactory.kt (hunk 1 above)
Comment (avidraghavMM) [reply app/src/main/java/com/raghav/spacedawn/ui/AppViewModelFactory.kt:12]: Dagger Hilt will be implemented for DI via a different PR

**Review by aditya-bhawsar-mm — APPROVED** [review]

**Pull request merged**